src/test/java/StringManipulationTest.java

```diff
@@ -55,27 +67,37 @@ public void testRemoveNthCharacter2() {
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello");
+        String result = manipulatedstring.removeNthCharacter(0, true);
+        assertEquals("Hello", result);
     }
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString(null);
+        String result = manipulatedstring.removeNthCharacter(3, true);
+        assertNull(result);
     }
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Testing");
+        String result = manipulatedstring.removeNthCharacter(2, true);
+        assertEquals("Tsting", result);
     }
 
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World!");
+        String result = manipulatedstring.removeNthCharacter(8, true);
+        assertEquals("Hello, Wold!", result);
     }
 
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Testing");
+        String result = manipulatedstring.removeNthCharacter(10, false);
+        assertEquals("Testing", result);
     }
 
```

Comment:
Another test case you could write would to have a negative number for N (such as manipulatedstring.setString(-1,false);
You could then assertThrows to throw an exception based on the feedback I provided in the StringManipulation file. This is just another edge case that could be encountered.